src/utils/IdealGraphVisualizer/ServerCompiler/src/main/java/com/sun/hotspot/igv/servercompiler/ServerCompilerScheduler.java

```diff
@@ -52,6 +53,38 @@
         public boolean isBlockProjection;
         public boolean isBlockStart;
         public boolean isCFG;
+        public int rank; // Rank for local scheduling priority.
+
+        public Node(InputNode n) {
+            inputNode = n;
+            String p = n.getProperties().get("is_block_proj");
+            isBlockProjection = (p != null && p.equals("true"));
```

Comment:
is it easy to flip equals here? 
"true".equals(p) can avoid nullcheck.

Reply:
Thanks for the tip! String comparisons are not a performance bottleneck in graph scheduling, and I find the current code more readable. But I will re-evaluate this suggestion after the main performance bottleneck in scheduling is removed in [JDK-8282043](https://bugs.openjdk.java.net/browse/JDK-8282043).